src/main/java/com/angelbroking/smartapi/http/SmartAPIResponseHandler.java

```diff
@@ -1,96 +1,87 @@
-import com.angelbroking.smartapi.http.exceptions.TokenException;
-
-import okhttp3.Response;
+import java.io.IOException;
 
 /**
  * Response handler for handling all the responses.
  */
 public class SmartAPIResponseHandler {
 
-	public JSONObject handle(Response response, String body) throws IOException, SmartAPIException, JSONException {
-		System.out.println("***************************");
-		if (response.header("Content-Type").contains("json")) {
-			JSONObject jsonObject = new JSONObject(body);
-			
-//			if (jsonObject.optString("data") == null || jsonObject.optString("data") == "") {
-			if (!jsonObject.has("status") || jsonObject.has("success")) {	
-				if (jsonObject.has("errorcode")) {
-					throw dealWithException(jsonObject, jsonObject.getString("errorcode"));
-				} else if (jsonObject.has("errorCode")) {
-					
-					throw dealWithException(jsonObject, jsonObject.getString("errorCode"));
-				}
-			}
-			//System.out.println(jsonObject);
-			return jsonObject;
-		} else {
-			throw new DataException("Unexpected content type received from server: " + response.header("Content-Type")
-					+ " " + response.body().string(), "AG8001");
-		}
-	}
-
-	private SmartAPIException dealWithException(JSONObject jsonObject, String code) throws JSONException {
-
-		switch (code) {
-		// if there is a token exception, generate a signal to logout the user.
-		case "AG8003":
-		case "AB8050":
-		case "AB8051":
-		case "AB1010":
-			if (SmartConnect.sessionExpiryHook != null) {
-				SmartConnect.sessionExpiryHook.sessionExpired();
-			}
-			return new TokenException(jsonObject.getString("message"), code);
-
-		case "AG8001":
-		case "AG8002":
-			return new DataException(jsonObject.getString("message"), code);
-
-		case "AB1004":
-		case "AB2000":
-			return new GeneralException(jsonObject.getString("message"), code);
-
-		case "AB1003":
-		case "AB1005":
-		case "AB1012":
-		case "AB1002":
-			return new InputException(jsonObject.getString("message"), code);
-
-		case "AB1008":
-		case "AB1009":
-		case "AB1013":
-		case "AB1014":
-		case "AB1015":
-		case "AB1016":
-		case "AB1017":
-			return new OrderException(jsonObject.getString("message"), code);
-
-		case "NetworkException":
-			return new NetworkException(jsonObject.getString("message"), code);
-
-		case "AB1000":
-		case "AB1001":
-		case "AB1011":
-			return new PermissionException(jsonObject.getString("message"), code);
-
-		default:
-			return new SmartAPIException(jsonObject.getString("data not found"));
-		}
-	}
+
+
+    public JSONObject handle(Response response, String body) throws IOException, SmartAPIException, JSONException {
+        if (response.header("Content-Type").contains("json")) {
+            JSONObject jsonObject = new JSONObject(body);
+
+            if (!jsonObject.has("status") || jsonObject.has("success")) {
+                if (jsonObject.has("errorcode")) {
+                    throw dealWithException(jsonObject, jsonObject.getString("errorcode"));

```

Comment:
Nested if's. This is not a production ready code.

Reply:
nested if's condition is removed an second if is added in below snippet to handle null pointer exception

            if (contentType == null || !contentType.contains("json")) {
                ResponseBody responseBody = response.body();
                if (responseBody != null) {
                    throw new DataSmartAPIException(String.format("Unexpected content type received from server: %s %s", response.header(CONTENT_TYPE), responseBody.string()), "AG8001");
                } else {
                    throw new DataSmartAPIException("Response body is null", "AG8001");
                }
            }
            JSONObject